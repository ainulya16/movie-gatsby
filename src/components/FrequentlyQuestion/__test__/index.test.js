import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { FrequentlyQuestion } from '../index';

describe('Frequently Asked Question', () => {
  it('render correctly faqs component', () => {
    const defaultProp = {
      faqData: {},
      onGetData: jest.fn(),
    };
    const component = renderer
      .create(<FrequentlyQuestion {...defaultProp} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
  it('should not show title and frequently asked question', () => {
    const defaultProp = {
      faqData: {},
      onGetData: jest.fn(),
    };
    const component = mount(<FrequentlyQuestion {...defaultProp} />);
    expect(component.find('h2')).toHaveLength(1);
  });
  it('should show correct title', () => {
    const props = {
      faqData: {
        title: 'title mock',
        data: [],
      },
      onGetData: jest.fn(),
    };
    const component = mount(<FrequentlyQuestion {...props} />);
    expect(component.find('h2')).toHaveLength(1);
    expect(
      component
        .find('h2')
        .first()
        .text(),
    ).toEqual('title mock');
  });
  it('should show correct frequently asked question data', () => {
    const props = {
      faqData: {
        title: 'title mock',
        data: [
          {
            id: 1,
            header: 'question mock',
            content: 'answer mock',
          },
        ],
      },
      onGetData: jest.fn(),
    };
    const component = mount(<FrequentlyQuestion {...props} />);
    expect(component.find('.ant-collapse-item')).toHaveLength(1);
    expect(
      component
        .find('.ant-collapse-header')
        .first()
        .text(),
    ).toEqual('question mock');
    // expect(component.find('.answer').first().text()).toEqual('answer mock');
  });
});
