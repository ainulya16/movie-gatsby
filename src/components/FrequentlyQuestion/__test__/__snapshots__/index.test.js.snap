// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Frequently Asked Question render correctly faqs component 1`] = `
.c9 {
  color: #000;
  font-size: 2em;
}

.c4 {
  width: 100%;
}

.c5 {
  border-bottom: solid 1px #dbb680;
  padding: 15px 0px;
}

.c7 {
  width: 50px;
  min-width: 50px;
  text-align: center;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c8 {
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: all 0.2s ease-out;
  transition: all 0.2s ease-out;
  overflow: hidden;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c6 p {
  font-size: 18px;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c1 {
  padding: 3.75em 7em;
}

.c1 h2 {
  font-family: 'AIABody';
  font-size: calc(18px + (26 - 14) * ((100vw - 300px) / (1600 - 300)));
  line-height: calc(1.3em + (1.5 - 1.2) * ((100vw - 300px) / (1600 - 300)));
  text-transform: uppercase;
  text-align: center;
  font-weight: bolder;
  margin-bottom: 30px;
}

.c2 {
  max-width: 687px;
  margin: 0 auto;
  font-family: 'AIAMedium';
}

.c2 h2 {
  color: #393d3e;
  font-family: 'AIACondensedMedium';
  font-size: 22px;
  font-weight: 500;
  -webkit-letter-spacing: 0.28px;
  -moz-letter-spacing: 0.28px;
  -ms-letter-spacing: 0.28px;
  letter-spacing: 0.28px;
  line-height: 28px;
  text-align: left;
}

.c2 h5 {
  color: #5798da;
  margin-bottom: 30px;
  cursor: pointer;
}

.c2 .ant-collapse {
  border-radius: 0px;
  background: none;
  border: none;
}

.c2 .ant-collapse .ant-collapse-item:last-child,
.c2 .ant-collapse .ant-collapse > .ant-collapse-item:last-child > .ant-collapse-header {
  border-radius: 0px;
  border-bottom: 1px solid #dbb680;
  background: none;
  height: 60px;
  min-width: 375px;
  max-width: 450px;
}

.c2 .ant-collapse .ant-collapse-item:last-child .ant-collapse-arrow,
.c2 .ant-collapse .ant-collapse > .ant-collapse-item:last-child > .ant-collapse-header .ant-collapse-arrow {
  margin-top: 10px;
}

.c2 .ant-collapse .ant-collapse-item > .ant-collapse-header {
  padding: 18px 0px;
}

.c2 .ant-collapse .ant-collapse-item > .ant-collapse-header svg {
  padding: 0px;
}

.c2 .ant-collapse-content {
  background: none;
}

.c2 .link {
  color: #685556;
  height: 22px;
  font-family: 'AIAMedium';
  font-size: 18px;
  font-weight: 100;
}

.c2 .downloadText {
  font-size: 18px;
  font-weight: 500;
  color: #685556;
  border-radius: 0;
  width: 70%;
  margin-left: 10px;
}

.c2 .headerText {
  font-family: 'AIAMedium';
  font-size: 18px;
  font-weight: 500;
  color: #685556;
}

.c2 .ant-collapse-content-box {
  font-family: 'AIARegular';
  font-size: 16ppx;
  line-height: 22px;
  -webkit-letter-spacing: 0.3px;
  -moz-letter-spacing: 0.3px;
  -ms-letter-spacing: 0.3px;
  letter-spacing: 0.3px;
  color: #333333;
}

.c2 .faqText > div {
  height: 60px;
  padding: 18px 0px;
  max-width: 450px;
  min-width: 375px;
}

.c2 .faqText > div > div > div {
  min-width: 32px;
  max-width: 40px;
  width: unset;
  text-align: left;
}

.c3 {
  -webkit-letter-spacing: 0.23px;
  -moz-letter-spacing: 0.23px;
  -ms-letter-spacing: 0.23px;
  letter-spacing: 0.23px;
  font-family: AIARegular;
  color: #333333;
  text-transform: unset;
}

.c10 {
  -webkit-letter-spacing: 0.23px;
  -moz-letter-spacing: 0.23px;
  -ms-letter-spacing: 0.23px;
  letter-spacing: 0.23px;
  font-family: AIAMedium;
  color: #333333;
  text-transform: unset;
}

@media (max-width:768px) {
  .c1 {
    padding: 2em 0;
  }
}

@media (max-width:768px) {
  .c1 h2 {
    margin-bottom: 20px;
  }
}

@media (max-width:1440px) {
  .c0 {
    padding: 60px 130px 35px 140px;
  }
}

@media (max-width:1024px) {
  .c0 {
    padding: 52.5px 98px;
  }
}

@media (max-width:768px) {
  .c0 {
    margin: auto;
  }
}

@media (max-width:426px) {
  .c0 {
    padding: 30px;
  }
}

@media (max-width:376px) {
  .c0 {
    padding: 30px 30px 30px 0px;
  }
}

@media (max-width:768px) {
  .c2 .ant-collapse > .ant-collapse-item:last-child > .ant-collapse-header {
    min-width: unset;
    width: 450px;
  }

  .c2 .faqText {
    width: 450px;
  }
}

<div
  className="c0 c1"
>
  <div
    className="c2"
  >
    <h2
      className="text-left c3"
      fontFamily="AIARegular"
    />
    <div
      className="faqText c4"
      style={null}
    >
      <div
        className="c5"
      >
        <div
          className="c6"
          onClick={[Function]}
        >
          <div
            className="c7"
          >
            <div
              className="c8"
              rotate={false}
            >
              <i
                className="icon-forward  c9"
                color="#000"
                size="m"
              />
            </div>
          </div>
          <p
            className="c10"
            fontFamily="AIAMedium"
          >
            <a
              className="link"
              href="/PA100/faqs"
            >
              Frequently Asked Questions (FAQs)
            </a>
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
